coreneuron/mpi/spdlog_mpi_logger.h

```diff
@@ -0,0 +1,92 @@
+/* Example
+#include <spdlog_mpi_logger.h>
+
+int main(int argc, char *argv[]) {
+    MPI_Init(&argc, &argv);
+    auto logger = std::make_shared<spdlog::mpi_logger>("my_mpi_logger");
+    logger->info("ceci est un test");
+    MPI_Finalize();
+    return 0;
+}
+*/
+
+#pragma once
+
+#include <iostream>
+#include <spdlog/logger.h>
+#include <mpi.h>
+
+namespace spdlog {
+
+class mpi_logger: public logger
+{
+public:
+    // Empty logger
+    explicit mpi_logger(std::string logger_name, int tag_ = 4444, int server_rank_ = 0, MPI_Comm comm_ = MPI_COMM_WORLD)
+        : logger(std::move(logger_name))
+        , tag(tag_), server_rank(server_rank_), comm(comm_)
+    {
+        MPI_Comm_rank(comm, &mpi_rank);
+        if (mpi_rank == server_rank) {
+            server_thr = std::thread([&](){
```

Comment:
few quick comments:
* I don't think we want to spawn the thread (performance & pinning aspects).
* another aspect is mixing two threading models - OpenMP in CoreNEURON and std::thread

Reply:
I think we can afford 1 thread in 1 rank before doing hours of computation.

Is it a problem as in this thread there is no OpenMP and that in OpenMP part there is no thread?

Reply:
(now I see where the slack discussion came from) I guess it's clear now that OpenMP _is_ threading. And I sort of agree with @pramodk that here the thread is not needed. It seems premature optimization and adding unnecessary complications to want to run the logging in its separate thread. I would say for now let's just skip threading.